pkg/utils/util_test.go

```diff
@@ -207,3 +207,130 @@ func TestRemovePrefixedCSIParams(t *testing.T) {
 		}
 	}
 }
+
+func TestShouldEnqueueContentChange(t *testing.T) {
+	oldValue := "old"
+	newValue := "new"
+
+	testcases := []struct {
+		name           string
+		old            *crdv1.VolumeSnapshotContent
+		new            *crdv1.VolumeSnapshotContent
+		expectedResult bool
+	}{
+		{
+			name:           "basic no change",
+			old:            &crdv1.VolumeSnapshotContent{},
+			new:            &crdv1.VolumeSnapshotContent{},
+			expectedResult: false,
+		},
+		{
+			name: "basic change",
+			old: &crdv1.VolumeSnapshotContent{
+				Spec: crdv1.VolumeSnapshotContentSpec{
+					VolumeSnapshotClassName: &oldValue,
+				},
+			},
+			new: &crdv1.VolumeSnapshotContent{
+				Spec: crdv1.VolumeSnapshotContentSpec{
+					VolumeSnapshotClassName: &newValue,
+				},
+			},
+			expectedResult: true,
+		},
+		{
+			name: "status change",
+			old: &crdv1.VolumeSnapshotContent{
+				Status: &crdv1.VolumeSnapshotContentStatus{
+					Error: &crdv1.VolumeSnapshotError{
+						Message: &oldValue,
+					},
+				},
+			},
+			new: &crdv1.VolumeSnapshotContent{
+				Status: &crdv1.VolumeSnapshotContentStatus{
+					Error: &crdv1.VolumeSnapshotError{
+						Message: &newValue,
+					},
+				},
+			},
+			expectedResult: false,
+		},
+		{
+			name: "managed annotation change",
+			old: &crdv1.VolumeSnapshotContent{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						AnnVolumeSnapshotBeingCreated: oldValue,
+					},
+				},
+			},
+			new: &crdv1.VolumeSnapshotContent{
+				ObjectMeta: metav1.ObjectMeta{
+					Annotations: map[string]string{
+						AnnVolumeSnapshotBeingCreated: newValue,
+					},
+				},
+			},
+			expectedResult: false,
```

Comment:
Can you add a test case where it changed from having AnnVolumeSnapshotBeingCreated to not having it and vice versa?